### DISCUSSION TITLE
Qt config tool for picom (picom-conf?)

### DISCUSSION THREAD

**Discussion by redtide** [lxqt/lxqt#2110] [Q&A] [answered]
I haven't used the Compton compositor for a while and the related Archlinux package has been replaced by Picom. I then saw that compton-conf was archived; out of curiosity, why there wasn't a "picom-conf"?

**Comment by tsujan** [comment, marked as answer]
LXQt treats all X11 compositors/WMs equally -- none receives a special treatment anymore.

**Reply by redtide** [reply]
OK, thank you!

**Reply by stefonarch** [reply]
Personally I would like to have a picom-conf in qt, atm the only way is using compton, configure settings and switch to picom.
But I think there is little hope somebody will work on it.

**Reply by tsujan** [reply]
It shouldn't be difficult to modify the code of `compton-conf` for that if the config keys are identical. If they aren't identical, more work will be needed. I just think it couldn't belong to LXQt.

**Reply by redtide** [reply]
I agree, it doesn't need to be part of the org, just an external utility, I'll try something myself. Maybe could some "adoption request" on the readme (or in here in discussions) be useful? Just an idea if there are some other people that would like to help on the task...

**Reply by tsujan** [reply]
I changed the title. It may attract the attention of those interested.

**Reply by redtide** [reply]
thank you!

**Comment by redtide** [comment]
I did some research [here](https://gist.github.com/redtide/f39f9958445d4e25d649742fef7eb2c7), it seems `compton-conf` could be ported with minor changes as a new `picom-conf` application.

**Reply by redtide** [reply]
For those who wants to give it a try/test I applied the changes [here](https://github.com/redtide/compton-conf/tree/picom-conf), and here a temporary [PKGBUILD](https://gist.github.com/redtide/ff3c905d56ec5ccca5622221663e6d71) for Archlinux.

**Reply by stefonarch** [reply]
Looks like it requires more work, adapting the keys:

[ 07/11/2021 17:46:53.266 parse_config_libconfig WARN ] Option `no-dock-shadow` is deprecated, and will be removed. Please use the wintype option `shadow` of `dock` instead.
[ 07/11/2021 17:46:53.266 parse_config_libconfig WARN ] Option `no-dnd-shadow` is deprecated, and will be removed. Please use the wintype option `shadow` of `dnd` instead.
```
[ 07/11/2021 17:46:53.266 parse_config_libconfig WARN ] Option `menu-opacity` is deprecated, and will be removed.Please use the wintype option `opacity` of `popup_menu` and `dropdown_menu` instead.
[ 07/11/2021 17:46:53.266 parse_config_libconfig WARN ] vsync option will take a boolean from now on. "none" is interpreted as "false" for compatibility, but this will stop working soon
[ 07/11/2021 17:46:53.266 parse_config_libconfig WARN ] glx-swap-method has been deprecated since v6, your setting "undefined" should be replaced by `use-damage = true`.
[ 07/11/2021 17:46:53.266 parse_config_libconfig WARN ] "clear-shadow" is removed as an option, and is always enabled now. Consider removing it from your config file
[ 07/11/2021 17:46:53.266 parse_config_libconfig ERROR ] "paint-on-overlay" has been removed as an option, and the feature is enabled whenever possible
[ 07/11/2021 17:46:53.266 main FATAL ERROR ] Failed to create new session.

```

**Reply by redtide** [reply]
My bad, sorry, I didn't pay enough attention on all options. I've updated the info in a [wiki page](https://github.com/redtide/compton-conf/wiki/TODO), the PKGBUILD, removed the deprecated/removed settings from the UI and replaced the config file with the new picom one. It should be at least an usable (even if scarce) start point now.

I wonder if I should adopt the project or is there some interest, maybe to put it in some "Qt utilities/goodies" organization?

**Reply by tsujan** [reply]
> I wonder if I should adopt the project

You could own the project; there's absolutely no problem with that. IMHO, you'll also be free to enhance it. People can always make PRs.

**Reply by redtide** [reply]
Thank you!

**Reply by stefonarch** [reply]
For the translations: we host also non-LXQt Qt apps but often combined by distros (qlipper, cmst, featherpad ecc) on our weblate instance, so ping me when it's ready.

**Reply by redtide** [reply]
cool, thank you! Well, I'm not sure at which point it could be considered ready, it has tons of features and the few ones are in seems not bad for me. I'm not sure when I'll be able to implement the wintypes option, which seems to be a "must have", for now I'm pushing the many and simple booleans. The repo is now renamed, I just have to merge the new branch (now temporarily the default one) into master.

**Reply by edwardp3** [reply]
Hi everyone. Due to issues with security updates, I removed Debian from my desktops and replaced it with (l)ubuntu.

On the Accessories menu, there are entries for both compton and picom. Selecting both of them, does not do anything, but running compton-conf in QTerminal brings up a GUI. I looked at it, did not change any settings, but changed the LXQt Theme from the default Lubuntu-Arc to Dark and unexpectedly had translucency in the menus, but when minimizing/maximizing a window, it was very erratic. When I rebooted, the translucency was gone.

**Reply by stefonarch** [reply]
Compton has a module in session settings. Picom has to be launched manually in autostart settings. Afaik picom replaces compton  when available both, You can also trying transparency in the panel settings, it will work only with a compositor running.

**Reply by edwardp3** [reply]
I turned on compton in Session Settings and logged back in. Got a grey screen briefly between the desktop manager/login screen and the LXQt desktop. Minimizing/maximizing the windows is still smooth, but had to turn off Smooth Scrolling in Firefox, as it was somewhat slow. Using the Dark theme, there is translucency without my having to change any settings.

**Reply by edwardp3** [reply]
I then turned compton off and did a logout/in. It was simply too slow. I could move the mouse pointer itself faster than moving a wnidow with the mouse.
